tasks/ruby.rb

```diff
@@ -3,44 +3,21 @@
 require 'open3'
 
 def facter_executable
-  if Gem.win_platform?
-    require 'win32/registry'
-    installed_dir =
-      begin
-        Win32::Registry::HKEY_LOCAL_MACHINE.open('SOFTWARE\Puppet Labs\Puppet') do |reg|
-          # rubocop:disable Style/RescueModifier
-          # Rescue missing key
-          dir = reg['RememberedInstallDir64'] rescue ''
-          # Both keys may exist, make sure the dir exists
-          break dir if File.exist?(dir)
+  install_path = File.join(File.dirname(RbConfig.ruby), 'facter')
 
-          # Rescue missing key
-          reg['RememberedInstallDir'] rescue ''
-          # rubocop:enable Style/RescueModifier
-        end
-      rescue Win32::Registry::Error
-        # Rescue missing registry path
-        ''
-      end
-
-    facter =
-      if installed_dir.empty?
-        ''
-      else
-        File.join(installed_dir, 'bin', 'facter')
-      end
+  # Fall back to PATH lookup if puppet-agent isn't installed
+  if File.exist?(install_path)
+    # Paths with spaces must be quoted on Windows, which means slashes need escaping
+    Gem.win_platform? ? "\"#{install_path.tr('/', '\\\\')}\"" : install_path
   else
-    facter = '/opt/puppetlabs/puppet/bin/facter'
+    'facter'
   end
-
-  # Fall back to PATH lookup if puppet-agent isn't installed
-  File.exist?(facter) ? facter : 'facter'
 end
 
 # Delegate to facter
 stdout, _stderr, _status = Open3.capture3("#{facter_executable} -v")
 if stdout =~ /^[0-2]\./
-  exec(facter_executable, '-p', '--json')
+  exec("#{facter_executable} -p --json")
```

Comment:
This ended up actually being an issue with how `exec` strips quotes from individual args, which causes the path to be passed 'bare' to the Windows shell which fails. Passing in a single string seems to fix this. Backticks were more reliable and consistent cross-platform, but it seems like most of the issues with `exec` and `system` on Windows were in Ruby 1.9, which we don't care about.